e2e/test/Helpers/TestDevice.cs

```diff
@@ -170,5 +176,25 @@ public async Task RemoveDeviceAsync()
             using var rm = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
             await rm.RemoveDeviceAsync(Id).ConfigureAwait(false);
         }
+
+        private static async Task RetryOnThrottlingAsync(Func<Task> asyncOperation)
```

Comment:
This could be a helper method available to all tests.. Would be useful whenever we want to retry Throttling exceptions.

Reply:
I agree, however, the reason I added it in here directly is because this is the only class in our test suite in which devices are created (outside of some specific jobs tests maybe). Provisioning clients can also create device identities but they are designed to handle throttling implicitly.

Reply:
Sure but when we need it again for something, we'll rewrite it. This implementation seems generic enough to reuse when needed (in case we find that we need to improve another test)

Reply:
I can add it to a helper class, no problem!